c/httpserver.c

```diff
@@ -3197,10 +3198,10 @@ static int serviceAuthNativeWithSessionToken(HttpService *service, HttpRequest *
      * We don't want to do this if we already found authentication data.
      */
     if (authDataFound == FALSE) {
-#define TLS_USERID_LENGTH 9
-      char userid[TLS_USERID_LENGTH] = {0};
       int racfReturnCode = 0, racfReasonCode = 0;
       zowelog(NULL, LOG_COMP_HTTPSERVER, ZOWE_LOG_DEBUG, "There was no token or credentials found in the request. Server is attempting to map the client certificate.\n");
+      userid = SLHAlloc(response->request->slh, 9);
+      memset(userid, 0, 9);
```

Comment:
Ooof.  That pointer from heap (regular) memory to stack memory,  `char userid[]`.   It's amazing it ever worked.  Really, how did it ever work?

Reply:
I think I was just lucky. I have the same thought...